### DISCUSSION TITLE
Reactive mode

### DISCUSSION THREAD

**Discussion by gsabbih6** [alandefreitas/matplotplusplus#69] [Q&A] [answered]
auto f = gcf(false);
auto ax = f->gca();
auto p = ax->plot(ax, x, y);   // draws once
p->color("red").line_width(2); // draws twice more
wait();

I am trying to use reactive mode for and interactive realtime graph but there is no gcf class that takes a false parameter in line 1 above. Any help?

This code was copied from the documentation directly

**Comment by alandefreitas** [comment, marked as answer]
The interface needed to change and I forgot to update the docs. You can try:

```cpp
figure_handle f = figure(true);
```

or

```cpp
figure_handle f = figure(false);
```

Like in the [geoplot_7.cpp](https://github.com/alandefreitas/matplotplusplus/blob/master/examples/geography/geoplot/geoplot_7.cpp) example.

You show also replace `wait()` with:

```cpp
show();
```

or

```cpp
draw();
```

If you need to update your figure very often, it'd be better to use the non-reactive mode and call draw() explicitly, like in the [geoplot_7.cpp](https://github.com/alandefreitas/matplotplusplus/blob/master/examples/geography/geoplot/geoplot_7.cpp) example.

I'll update the docs to fix that.

Thanks,

**Comment by gsabbih6** [comment]
Thanks alandefreitas
much appreciated
